src/ocean/util/container/queue/FixedRingQueue.d

```diff
@@ -196,7 +196,7 @@ class FixedByteRingQueue : FixedRingQueueBase!(IByteQueue)
 
         if (element_in_queue)
         {
-            element_in_queue[] = element[];
+            (cast(ubyte[])element_in_queue)[] = element[];
```

Comment:
This alone is a big no-no

Reply:
What do you suggest instead?

Reply:
For clarity: `typeof(element_in_queue)` is `void[]`.

Reply:
```D
void[] element_ = element[];
element_in_queue[] = element_[];
```

Reply:
This results in _Error: cannot implicitly convert expression `(element[])` of type `const(ubyte)[]` to `void[]`._

A detailed description of what I'm doing is
```d
// The function argument.
const(ubyte[]) element;
// Get the copy destination buffer, its type is mutable `void[]`.
void[] element_in_queue = super.push_();
// Cast the copy destination buffer from mutable `void[]` to mutable `ubyte[]`.
ubyte[] element_in_queue_ = cast(ubyte[])element_in_queue;
// Copy the array content of `element` into the destination buffer.
element_in_queue_[] = element[];
```
In the actual code I do the last two steps in one line:
```d
(cast(ubyte[])element_in_queue)[] = element[];
```
I don't see which of these steps is wrong.

Reply:
`Const!(void)[] element_ = element[]` in that case.

What you are doing wrong is using a cast. Every single cast is an armed nuclear warhead - if you can avoid having one in your cupboard, you probably should do so.

Reply:
I see, thanks for the clarification.